ansible/roles/test/tasks/ptf_runner.yml

```diff
@@ -51,5 +51,36 @@
 
 - debug: var=out.stdout_lines
 
+- name: Set default PTF log filename
+  set_fact:
+    ptf_log_file: "/root/ptf.log"
+    ptf_log_file_param_index: "{{ out.cmd.find('--log-file') }}"
+
+- name: Parse custom log filename specified in PTF command
+  set_fact:
+    ptf_log_file: "{{ out.cmd[ptf_log_file_param_index|int:].split(' ')[1] }}"
+  when: ptf_log_file_param_index|int >= 0
+
+- name: Set PTF pcap filename
+  set_fact:
+    ptf_pcap_file: "{{ ptf_log_file | replace('.log', '.pcap') }}"
+
+- name : Fetch result files from switch to ansible machine
+  fetch:
+    src: "{{ item }}"
+    dest: "test/{{ inventory_hostname }}/ptf/{{ item | basename }}.{{lookup('pipe','date +%Y-%m-%d-%H:%M:%S')}}"
+    flat: yes
+  with_items:
+    - "{{ ptf_log_file }}"
+    - "{{ ptf_pcap_file }}"
+  delegate_to: "{{ ptf_host }}"
+  when: out.rc != 0
```

Comment:
Let's save the logs in all cases.

Reply:
Since there is no auto cleanup or rotate mechanism for saved PTF log and pcap files, storage on sonic-mgmt container could be quickly used up if we save logs in all cases.

If we have interest in log/pcap of certain successful PTF cases, we can always check them in the PTF container before the container is destroyed.

Do you still think it is necessary to save the logs in all cases? @pavel-shirshov

Reply:
My opinion it's better to have all information saved from the test: both successes and failures.
I think we have filenames the same. So every test will overwrite files.

Reply:
The filenames would not be same. To avoid overwriting, I added code to append a timestamp to each file saved to sonic-mgmt.

Reply:
I see. In MS case we have Jenkins which run the test cases for us.
Later Jenkins copies all the necessary files to the Jenkins server.
So in our case we're not interested to timestamping filenames. Jenkins does that for us

Reply:
Maybe I can add another parameter to specify whether failed, successful, or both logs should be saved?

Reply:
let's keep it as it is.